core/src/main/java/org/apache/iceberg/RemoveSnapshots.java

```diff
@@ -352,7 +352,7 @@ public void commit() {
             });
     LOG.info("Committed snapshot changes");
 
-    if (cleanExpiredFiles) {
+    if (cleanExpiredFiles && base.currentSnapshot() != null) {
```

Comment:
i think I will need to fix this, it's possible that the metadata before expiration has no main but has other files to clean up!